### PR TITLE
Python: Add modeling of simplejson/ujson/idna

### PR DESCRIPTION
**Pull request by RasmusWL**
These could all have gotten their own PR, but I thought they were similar enough to just keep it as one PR :shrug:

### CODE REVIEW

File: python/ql/src/semmle/python/frameworks/Ujson.qll [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  /**
+   * A call to `ujson.dump`.
+   */
+  private class UjsonDumpCall extends Encoding::Range, DataFlow::CallCfgNode {
+    UjsonDumpCall() { this = API::moduleImport("ujson").getMember("dump").getACall() }
+
+    override DataFlow::Node getAnInput() { result = this.getArg(0) }
+
+    override DataFlow::Node getOutput() {
+      result.(DataFlow::PostUpdateNode).getPreUpdateNode() = this.getArg(1)
```
Comment (tausbn) [inline python/ql/src/semmle/python/frameworks/Ujson.qll:39]: I feel like this is clunky enough that we should extend the API to make it easy to access the `PostUpdateNode` (if it it exists) for a given `Node`. 

That is, something like
```ql
PostUpdateNode getPostUpdateNode() { result.getPreUpdateNode() = this } 
```
on the `Node` class itself, so we could just write `result = this.getArg(1).getPostUpdateNode()` above.

**Review by tausbn — APPROVED** [review]
Looks good to me. I have a single comment but it's somewhat outside the scope of this PR, and probably best left for a different PR. Accordingly, I have marked this PR as approved. :+1:

File: python/ql/src/semmle/python/frameworks/Ujson.qll (hunk 1 above)
Comment (RasmusWL) [reply python/ql/src/semmle/python/frameworks/Ujson.qll:39]: That would be nice indeed :+1:

**Comment by RasmusWL** [comment]
:shipit:

File: python/ql/src/semmle/python/frameworks/Ujson.qll (hunk 1 above)
Comment (RasmusWL) [reply python/ql/src/semmle/python/frameworks/Ujson.qll:39]: I did this in a PR: https://github.com/github/codeql/pull/5926

**Pull request merged**